### PR TITLE
Call function

### PR DESCRIPTION
**Pull request by madammann**
The pull request for the call function is out!

Currently I am only 80% sure what I did is correct so all the feedback I can get is appreciated.
I had to rewrite parts of the evaluation function as well since "p.err" was never changed and I was sure, reading the PDF from the coding session, that p.err was meant to be used in the call function update.

Please check my changes to the evaluation function and to the call function and tell me whether they make sense.
Also please feel free to do commits on this branch too if you figure out an issue and can solve it.

For the visualization at the end and the results, since only 2 iterations are shown the changes are not major, however, from what I tested they worked correctly but with pretty small position changes per iteration.

Best,
Marlon

### CODE REVIEW

**Comment by madammann** [comment]
I am also not sure whether this branch should be merged with coding_support_results or particle swarm, feel free to change the base accordingly.

**Review by FaKock — APPROVED** [review]
Hello Marlon, 

Thanks for being so fast! 

The code seems to be in working order to me and could be executed without problem. The only thing I would maybe add is a print() before the visualization making it clear that it is the final or the n-th visualization. I did not really see that directly in the output once I tried to run the code, but maybe that's just me. Or as you mentioned it is due to the only small changes.

Thanks for correcting the mistake in the evaluation function. We did not notice it during the review.
Merge in the main particle swarm is correct. If the update functions work as intended, then I'll later on just merge this branch into particle swarm and delete the other pull request. 

I'll wait for another review or two and then merge the request and give Inga the go.

Have a nice weekend!

**Pull request merged**
